src/tree/RefreshManager.ts

```diff
@@ -0,0 +1,169 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzExtTreeItem, callWithTelemetryAndErrorHandling, IActionContext, parseError, registerCommand } from '@microsoft/vscode-azext-utils';
+import * as vscode from 'vscode';
+import { ext } from '../extensionVariables';
+import { localize } from '../localize';
+import { isCancellationError } from '../runtimes/docker';
+import { AllTreePrefixes, TreePrefix } from './TreePrefix';
+
+const pollingIntervalMs = 60 * 1000; // One minute
+const eventListenerTries = 3; // The event listener will try at most 3 times to connect for events
+
+type RefreshTarget = AzExtTreeItem | TreePrefix;
+type RefreshReason = 'interval' | 'event' | 'config' | 'manual';
+
+export class RefreshManager extends vscode.Disposable {
+    private readonly refreshEmitter = new vscode.EventEmitter<[RefreshTarget, RefreshReason]>();
+    private readonly disposables: vscode.Disposable[] = [];
+    private readonly cts: vscode.CancellationTokenSource = new vscode.CancellationTokenSource();
+
+    public constructor() {
+        super(() => vscode.Disposable.from(...this.disposables).dispose());
+
+        this.disposables.push(
+            this.refreshEmitter.event(async ([target, reason]) => {

```

Comment:
This is needlessly complex, an event isn't necessary to forward to an always-singleton listener.